src/Coxeter_triangulation/include/gudhi/IO/build_mesh_from_cell_complex.h

```diff
@@ -0,0 +1,167 @@
+      ci_map.emplace(std::make_pair(cell, index++));
+      output.vertex_points.emplace_back((1. / vertex_indices.size()) * barycenter);
+#ifdef DEBUG_TRACES
+      std::string vlist = " (" + std::to_string(index - 1) + ")";
+      for (const std::size_t& v : vertex_indices) vlist += " " + std::to_string(v);
+      cell_vlist_map.emplace(std::make_pair(to_string(cell), vlist));
+#endif
+    }
+
+  if (configuration.toggle_edges && sc_map.size() >= 2)
+    for (const auto& sc_pair : sc_map[1]) {
+      Hasse_cell* edge_cell = sc_pair.second;
+      Mesh_element_vertices edge;
+      for (const auto& vi_pair : edge_cell->get_boundary()) edge.push_back(vi_map[vi_pair.first]);
+      output.edges.emplace_back(std::make_pair(edge, configuration.ref_edges));
+#ifdef DEBUG_TRACES
+      std::string vlist;
+      for (const std::size_t& v : edge) vlist += " " + std::to_string(v);
+      cell_vlist_map.emplace(std::make_pair(to_string(edge_cell), vlist));
+#endif
+    }
+
+  if (configuration.toggle_triangles && sc_map.size() >= 3)
+    for (const auto& sc_pair : sc_map[2]) {
+      for (const auto& ei_pair : sc_pair.second->get_boundary()) {
+        Mesh_element_vertices triangle(1, ci_map[sc_pair.second]);
+        for (const auto& vi_pair : ei_pair.first->get_boundary()) triangle.push_back(vi_map[vi_pair.first]);
+        output.triangles.emplace_back(std::make_pair(triangle, configuration.ref_triangles));
+      }
+    }
+
+  if (configuration.toggle_tetrahedra && sc_map.size() >= 4)
+    for (const auto& sc_pair : sc_map[3]) {
+      Eigen::VectorXd barycenter = Eigen::VectorXd::Zero(amb_d);
+      std::set<std::size_t> vertex_indices;
+      Hasse_cell* cell = sc_pair.second;
+      for (const auto& ci_pair : cell->get_boundary())
+        for (const auto& ei_pair : ci_pair.first->get_boundary())
+          for (const auto& vi_pair : ei_pair.first->get_boundary()) vertex_indices.emplace(vi_map[vi_pair.first]);
+      for (const std::size_t& v : vertex_indices) barycenter += output.vertex_points[v - 1];
+      output.vertex_points.emplace_back((1. / vertex_indices.size()) * barycenter);
+#ifdef DEBUG_TRACES
+      std::string vlist = " (" + std::to_string(index) + ")";
+      for (const std::size_t& v : vertex_indices) vlist += " " + std::to_string(v);
+      cell_vlist_map.emplace(std::make_pair(to_string(cell), vlist));
+#endif
+
+      for (const auto& ci_pair : cell->get_boundary())
+        for (const auto& ei_pair : ci_pair.first->get_boundary()) {
+          Mesh_element_vertices tetrahedron = {index, ci_map[sc_pair.second]};
+          for (const auto& vi_pair : ei_pair.first->get_boundary()) tetrahedron.push_back(vi_map[vi_pair.first]);
+          output.tetrahedra.emplace_back(std::make_pair(tetrahedron, configuration.ref_tetrahedra));
+        }
+      index++;
+    }
+}
+
+template <class Cell_complex>
+Mesh_medit build_mesh_from_cell_complex(const Cell_complex& cell_complex,
```

Comment:
No documentation?

Reply:
doc review: fix `build_mesh_from_cell_complex` and `output_meshes_to_medit` doxygen issue on af94bf9d